### ISSUE TITLE
Ease of use enhancement

### ISSUE THREAD

**Issue by aldus2** [rgertenbach/Lua-Stats#4]
New here so I apologize for any failure in the local etiquette.
Thanks for this useful package of useful primitives.
I spent some times integrating this "library" and the result is not completely satisfactory.
May I suggest two enhancements :

- an example of the way(s) to integrate this "library" into a project (copy functions, dofile, require, ...) would help
- it would be nice to make it a real lua library, accessible after a local stats=require('lua-stats') and with var1=stats.pNorm(...) for instance, following the model with math, var1=math.sin(...).
I wish I knew to do the transformation and contribute, but I don't right now :(
Ready to help where I can !

**Comment by rgertenbach** [comment]
Hey, thanks for commenting.

tl;dr I'd recommend using a different library such as SciLua.

In general I wouldn't really recommend using it, for a number of reasons:
- I used this in a project by copy and pasting it, it's my first and last Lua project.
- It's not tested beyond close-enough-checks before committing, I didn't expect anyone to ever see this.
- It's slow, e.g. I don't see a reason for factorial to be recursive for integers when it can be so slow and an alternative is programmed, the integration is stepwise instead of using more sophisticated method like gradients
- That also tends to make it less exact for a given runtime

I absolutely do not mind if you make your own version using any subset or all of this code but as mentioned cannot guarantee the usefulness or correctness.

**Comment by aldus2** [comment]
Thanks for tour feedback !

Aldus
(mobile)

Le 16 mai 2021 à 23:45, Robin Gertenbach ***@***.***> a écrit :

﻿

Hey, thanks for commenting.

tl;dr I'd recommend using a different library such as SciLua.

In general I wouldn't really recommend using it, for a number of reasons:

  *   I used this in a project by copy and pasting it, it's my first and last Lua project.
  *   It's not tested beyond close-enough-checks before committing, I didn't expect anyone to ever see this.
  *   It's slow, e.g. I don't see a reason for factorial to be recursive for integers when it can be so slow and an alternative is programmed, the integration is stepwise instead of using more sophisticated method like gradients
  *   That also tends to make it less exact for a given runtime

I absolutely do not mind if you make your own version using any subset or all of this code but as mentioned cannot guarantee the usefulness or correctness.
